scripts/west_commands/run_common.py

```diff
@@ -201,6 +322,84 @@ def do_run_common_image(command, user_args, user_runner_args, build_dir=None):
     # parsing, it will show up here, and needs to be filtered out.
     runner_args = [arg for arg in user_runner_args if arg != '--']
 
+    # Check if there are any commands that should only be ran once per board
+    # and if so, remove them for all but the first iteration of the flash
+    # runner per unique board name.
+    if len(used_cmds) > 0 and len(runner_args) > 0:
+        l = len(runner_args) - 1
+        while l >= 0:
+            i = 0
+            while i < len(used_cmds):
+                if used_cmds[i].command == runner_args[l] and (runner_name in used_cmds[i].runners \
+                                                               or 'all' in used_cmds[i].runners):
+                    # Check if board is here
+                    found = False
+                    for check in used_cmds[i].boards:
+                        if board == check:
+                            found = True
+                            break
```

Comment:
This is checking: `board in used_cmds[i].boards`

Reply:
Fixed